subsys/net/ip/6lo.c

```diff
@@ -756,7 +756,7 @@ static inline bool compress_IPHC_header(struct net_pkt *pkt,
 
 	/* UDP header compression */
 	if (IS_ENABLED(CONFIG_NET_UDP)) {
-		struct net_udp_hdr hdr, *udp;
+		struct net_udp_hdr hdr __unused, *udp;
```

Comment:
You mean changing both the function definition and macro? The macro itself is only used when CONFIG_NET_UDP is not defined.